Source/SwiftLintCore/Extensions/NSRegularExpression+SwiftLint.swift

```diff
@@ -4,6 +4,26 @@ import SourceKittenFramework
 private var regexCache = [RegexCacheKey: NSRegularExpression]()
 private let regexCacheLock = NSLock()
 
+public struct RegularExpression: Hashable, Comparable, ExpressibleByStringLiteral {
```

Comment:
Oh, I thought when I saw references to this elsewhere it was referring to [`Regex`](https://developer.apple.com/documentation/swift/regex).

Reply:
One cannot ask `Regex` for the underlying `pattern` which we'd need for documentation. That's why I went with this custom value-type for regular expressions.

Reply:
We could easily have a wrapper type that keeps a copy of the `pattern` that we created a `Regex` with for documentation purposes, but this is unrelated to this PR.